// @expo/prebuild-config/build/plugins/unversioned/expo-system-ui/withIosUserInterfaceStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getUserInterfaceStyle = getUserInterfaceStyle;
exports.setUserInterfaceStyle = setUserInterfaceStyle;
exports.withIosUserInterfaceStyle = void 0;
<<<<<<<< HEAD:learningkidsapp/node_modules/@expo/cli/node_modules/@expo/prebuild-config/build/plugins/unversioned/expo-system-ui/withIosUserInterfaceStyle.js
function _iosPlugins() {
  const data = require("@expo/config-plugins/build/plugins/ios-plugins");
  _iosPlugins = function () {
========

function _configPlugins() {
  const data = require("@expo/config-plugins");

  _configPlugins = function () {
>>>>>>>> 279c076d3cb6520341221d8adc510e9dbdcbb7c2:learningkidsapp/node_modules/expo-splash-screen/node_modules/@expo/prebuild-config/build/plugins/unversioned/expo-system-ui/withIosUserInterfaceStyle.js
    return data;
  };
  return data;
}
<<<<<<<< HEAD:learningkidsapp/node_modules/@expo/cli/node_modules/@expo/prebuild-config/build/plugins/unversioned/expo-system-ui/withIosUserInterfaceStyle.js
const withIosUserInterfaceStyle = exports.withIosUserInterfaceStyle = (0, _iosPlugins().createInfoPlistPluginWithPropertyGuard)(setUserInterfaceStyle, {
  infoPlistProperty: 'UIUserInterfaceStyle',
  expoConfigProperty: 'userInterfaceStyle | ios.userInterfaceStyle',
  expoPropertyGetter: getUserInterfaceStyle
}, 'withIosUserInterfaceStyle');
========

const withIosUserInterfaceStyle = config => {
  return (0, _configPlugins().withInfoPlist)(config, config => {
    config.modResults = setUserInterfaceStyle(config, config.modResults);
    return config;
  });
};

exports.withIosUserInterfaceStyle = withIosUserInterfaceStyle;

>>>>>>>> 279c076d3cb6520341221d8adc510e9dbdcbb7c2:learningkidsapp/node_modules/expo-splash-screen/node_modules/@expo/prebuild-config/build/plugins/unversioned/expo-system-ui/withIosUserInterfaceStyle.js
function getUserInterfaceStyle(config) {
  return config.ios?.userInterfaceStyle ?? config.userInterfaceStyle ?? 'light';
}
function setUserInterfaceStyle(config, {
  UIUserInterfaceStyle,
  ...infoPlist
}) {
  const userInterfaceStyle = getUserInterfaceStyle(config);
  const style = mapUserInterfaceStyleForInfoPlist(userInterfaceStyle);
  if (!style) {
    return infoPlist;
  }
  return {
    ...infoPlist,
    UIUserInterfaceStyle: style
  };
}
function mapUserInterfaceStyleForInfoPlist(userInterfaceStyle) {
  switch (userInterfaceStyle) {
    case 'light':
      return 'Light';
    case 'dark':
      return 'Dark';
    case 'automatic':
      return 'Automatic';
  }
  return null;
}
